src/tools/miri/src/bin/miri.rs

```diff
@@ -372,6 +372,9 @@ fn main() {
     let using_internal_features =
         rustc_driver::install_ice_hook("https://github.com/rust-lang/miri/issues/new", |_| ());
 
+    // Install the ctrlc handler that sets `rustc_const_eval::CTRL_C_RECEIVED`.
+    rustc_driver::install_ctrlc_handler();
```

Comment:
This should probably be *before* the `MIRI_BE_RUSTC` case distinction, otherwise no handler is set when Miri acts as rustc.